### PR TITLE
class target for annotation, log all methods

### PR DESCRIPTION
**Pull request by RomainPiel**
With this, annotating the class does exactly the same as annotating each method:

```java
  @DebugLog
  static class Charmer {
    private final String name;

    Charmer(String name) {
        this.name = name;
    }

    public String askHowAreYou() {
        return "How are you " + name + "?";
    }
  }
```

will log:
```
               Charmer  V  ⇢ <init>(name="Jake")
                        V  ⇠ <init> [0ms]
                        V  ⇢ askHowAreYou()
                        V  ⇠ askHowAreYou [0ms] = "How are you Jake?"
```

Calling super methods/constructors will only log if the parent class is annotated or if the methods/constructors are annotated.

### CODE REVIEW

File: hugo-example/src/main/java/com/example/hugo/HugoActivity.java [hunk 1]
```diff
@@ -70,4 +73,17 @@ public String sayHello() {
...
+  @DebugLog
+  static class Charmer {
+    private final String name;
+
+    Charmer(String name) {
+        this.name = name;
+    }
+
+    public String askHowAreYou() {
+        return "How are you " + name + "?";
```
Comment (JakeWharton) [inline hugo-example/src/main/java/com/example/hugo/HugoActivity.java]: super nit: this and constructor should only be two-space indented

**Comment by JakeWharton** [comment]
Awesome. One tiny nit and I'll get this in. And hey, it'll motivate me enough to make a release (finally).

**Comment by RomainPiel** [comment]
Cool! Worth mentioning the original issue: https://github.com/JakeWharton/hugo/issues/1

**Comment by JakeWharton** [comment]
Squashed and merged. Thanks!

**Comment by JakeWharton** [comment]
Release soon.

**Pull request closed without merging**